x-pack/plugins/apm/server/lib/services/get_services/index.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isEmpty } from 'lodash';
+import { PromiseReturnType } from 'x-pack/plugins/apm/typings/common';
+import { Setup } from '../../helpers/setup_request';
+import { getAgentStatus } from './get_agent_status';
+import { getLegacyDataStatus } from './get_legacy_data_status';
+import { getServicesItems } from './get_services_items';
+
+export type ServiceListAPIResponse = PromiseReturnType<typeof getServices>;
+export async function getServices(setup: Setup) {
+  const items = await getServicesItems(setup);
+  const hasLegacyData = await getLegacyDataStatus(setup);
+
+  // conditionally check for historical data if no services were found in the current time range
+  const noDataInCurrentTimeRange = isEmpty(items);
+  let hasHistorialAgentData = true;
+  if (noDataInCurrentTimeRange) {
+    hasHistorialAgentData = await getAgentStatus(setup);
+  }
```

Comment:
So did we decide to only do this extra request and notification for the service list query? And was it not possible or too complicated to use the `msearch` approach vs two separate requests from the front end? Just curious to hear what you found as you implemented, this all looks good.

Reply:
Yes, It became a bit more convoluted to use msearch here, and since it's not a hotpath I chose to more or less keep it like it used to be.

I'm pretty sure the overall request will end up being faster for couple of reasons:
 - we used to get the agent status unconditionally: now we only get it if no data was found
 - agent status was fetched as a separate request from the browser. Not they are at least batched in Node, which is physically much closer to ES than the browser is. Using `msearch` would further optimize this but I don't think it's strictly necessary. 
 - I added `terminateAfter: 1` to the agent status request, so it only returns a single doc from each shard (instead of counting all of them).